### PR TITLE
fix(parser): clamp PHP require-target resolution to workspaceRoot (#1009)

### PR DESCRIPTION
**Pull request by alfhen**
## Summary

Small follow-up to #1063 (merged). CodeRabbit's review on that PR landed a valid finding just after the merge went through — reported here rather than left unaddressed.

`requireTargetExists` (`packages/parser/src/php-require.ts`) used `path.join(workspaceRoot, specifier)` to check whether a statically-resolved PHP `require`/`include` target exists on disk. `path.join` resolves `..` segments but does **not** clamp the result to stay inside `workspaceRoot` — so a specifier like `require __DIR__ . '/../../../../etc/passwd';` (statically resolvable by #1063's own rules) could `fs.statSync` a real file entirely outside the indexed project and get trusted as a genuine require target.

Note this is *not* a new category of behavior for the codebase: JS/TS/Python's own `resolveRelativeImport` already permits a specifier to climb outside `workspaceRoot` as a plain string (accepted, tested — see `path-matching.test.ts`'s `'../../../../outside/thing'` case). What's different here is that `requireTargetExists` is the only place in that whole resolution pipeline that actually touches the real filesystem for the resolved result, which is what makes the boundary escape observable (a real `stat` call against a path outside the project) rather than just an inert string that never matches anything.

## Fix

`requireTargetExists` now resolves the candidate path, computes its relative path back to `workspaceRoot`, and rejects (returns `false`) if that relative path escapes (starts with `..`, or is itself absolute — the cross-drive case on Windows) *before* ever calling `fs.statSync`.

## Test

Added a regression test: creates a sibling directory under `os.tmpdir()` (alongside the test's own `workspaceRoot` temp dir) with a real file in it, then asserts a `../<sibling>/secret.php`-shaped specifier returns `false` even though the file genuinely exists. Verified this test fails without the boundary check (reproduces the exact escape) and passes with it.

## Gates

- `npm run format:check` — pass
- `npm run lint` — 0 errors
- `npm run typecheck` — pass
- `npm run build` — pass
- `npm test` — pass (full monorepo suite, 3745 tests across all packages)
- `lien delta --base origin/main` — exit 0 (`requireTargetExists` cognitive complexity 2 → 4, well under threshold)

Scope: touches only `php-require.ts`/`php-require.test.ts`, both introduced by #1063. No behavior change for any specifier that resolves inside `workspaceRoot` (the overwhelming common case).

---

### Lien Review

✅ **Good** - No complexity issues found.

> [!NOTE]
> **Low Risk**
>
> This PR clamps PHP require-target resolution to workspaceRoot, preventing path-traversal stat calls outside the project. The fix correctly uses path.resolve + path.relative and checks for an actual parent-segment escape (`..` or `..${path.sep}`) rather than a naive `startsWith('..')` substring match, avoiding the over-rejection of legitimate files whose names begin with two dots. The added regression tests cover the escape case, the `..foo.php` edge case, nested legitimate paths, and internal `..` segments that net inside workspaceRoot. Callers in ast/symbols.ts pass workspaceRoot consistently and are unaffected for in-project specifiers.
>
> - requireTargetExists now resolves and boundary-checks the candidate path before calling fs.statSync
> - Boundary check correctly distinguishes parent-directory escapes from filenames starting with `..`
> - Added regression tests for traversal escape and legitimate in-project edge cases

✅ *Trust: **Delivered** — The review ran to completion within budget.*

<sup>Reviewed by [Lien Review](https://lien.dev) for commit 1bce32f. Updates automatically on new commits.</sup>

### CODE REVIEW

File: packages/parser/src/php-require.ts [hunk 1]
```diff
@@ -22,16 +22,35 @@ import path from 'path';
...
 export function requireTargetExists(specifier: string, workspaceRoot: string | undefined): boolean {
   if (!workspaceRoot) return false;
+
+  const resolvedRoot = path.resolve(workspaceRoot);
+  const candidate = path.resolve(resolvedRoot, specifier);
+  const relativeToRoot = path.relative(resolvedRoot, candidate);
+  // A specifier that resolves outside workspaceRoot produces a relative path
+  // starting with '..' (or, if the two are on different Windows drives, an
+  // absolute path) -- either way, never trust it as a real project file.
+  if (relativeToRoot.startsWith('..') || path.isAbsolute(relativeToRoot)) return false;
```
[the remark this answers is not in the document]
Comment (alfhen) [reply packages/parser/src/php-require.ts]: Confirmed and fixed exactly as suggested: the check is now relativeToRoot === '..' || relativeToRoot.startsWith(`..${path.sep}`) (a real parent-segment escape) rather than a bare startsWith('..') substring match. Added a test for a file literally named '..foo.php' inside workspaceRoot -- verified it fails against the previous version (reproduces the over-rejection) and passes with this fix. Fixed in commit 1bce32fb.
